### ISSUE TITLE
amp-ad: Language of page content is not indicated

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#29562]
## WCAG Level
Level A

## Priority
Low

## Type
Structure

## Pages/screens/Components affected
* [amp-ad](https://amp.dev/documentation/components/amp-ad/?format=websites)

## Description
It is important to programmatically identify the human language of a web page/section of a page, as this makes a difference as to how screen readers pronounce content.

In the examples provided in [Glitch](https://daisy-purring-seal.glitch.me/amp-ad.html) and the [playground](https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Fcomponents%2Famp-ad&format=websites&_gl=1*6tq095*_ga*YW1wLWdDQmZ0cDgwZl96MWxoRWR1NEFlaXc.) the advert that is embedded in the `<iframe>` that is generated (using an `<amp-ad ... type="A9">`) lacks any kind of language definition in its markup. Note that because the overall page lacks a programmatically defined language, which fails WCAG 2.1 SC 3.1.1 Language of the Page, this also leads to the language of all *parts* of the content also lacking a programmatically defined language, failing WCAG 2.1 SC 3.1.2 Language of Parts as well.

## User impact
Depending on the user’s setup, screen readers and other text-to-speech applications that support multiple languages will not be able to adapt to the correct accent or pronunciation when announcing page content, which can impact on the understandability of the page.

## Required solution
Make sure that the human language of all HTML pages, as well as content embedded within pages and changes in language, is declared correctly. In the case of `<iframe>` elements, the content inside the embedded document does not automatically inherit the language definition from its host page - make sure that `<html>` elements of the content that is embedded includes the required `lang` attribute, and that the value of this attribute matches the human language of the ad content.

This content is likely provided by the third-party ad networks. If at all possible, these providers should be encouraged/told that their current markup lacks an appropriate definition for the language of their content, such as

```
<html lang="en">
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied:
1. Check that the `<html>` element on the ad document embedded in the `<iframe>` once the `<amp-ad>` element is rendered has a `lang` attribute.
2. Check that the value of the `lang` attribute reflects the primary language used by the ad content.

## Related standards
* [WCAG 2.1 Success Criterion 3.1.1 Language of Page (Level A)](https://www.w3.org/TR/WCAG21/#language-of-page)
* [WCAG 2.1 Success Criterion 3.1.2 Language of Parts (Level AA)](https://www.w3.org/TR/WCAG21/#language-of-parts)

## More information
* [W3C - Technique H57: Using the language attribute on the HTML element](https://www.w3.org/WAI/WCAG21/Techniques/html/H57.html)
* [W3C - Technique H58: Using language attributes to identify changes in the human language](https://www.w3.org/WAI/WCAG21/Techniques/html/H58.html)

## Test data
Test date: July 2020

**Comment by caroqliu** [comment]
FYI @ampproject/wg-ads
